scylla-cdc/src/reader.rs

```diff
@@ -0,0 +1,340 @@
+        format!("CREATE TABLE IF NOT EXISTS {}.{} (pk int, t int, v text, s text, PRIMARY KEY (pk, t)) WITH cdc = {{'enabled':true}};",
+                TEST_KEYSPACE,
+                TEST_TABLE
+        )
+    }
+
+    async fn create_test_db(session: &Session) -> Result<(), Box<dyn Error>> {
+        let mut create_keyspace_query = Query::new(format!(
+            "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class': 'SimpleStrategy', 'replication_factor': 1}};",
+            TEST_KEYSPACE
+        ));
+        create_keyspace_query.set_consistency(Consistency::All);
+
+        session.query(create_keyspace_query, &[]).await?;
+        session.await_schema_agreement().await?;
+
+        // Create test table
+        let create_table_query = get_create_table_query();
+        session.query(create_table_query, &[]).await?;
+        session.await_schema_agreement().await?;
+
+        let table_name_under_keyspace = format!("{}.{}", TEST_KEYSPACE, TEST_TABLE);
+        session
+            .query(format!("TRUNCATE {};", table_name_under_keyspace), &[])
+            .await?;
+        Ok(())
+    }
+
+    async fn populate_test_db(session: &Session) -> Result<(), Box<dyn Error>> {
+        let table_name_under_keyspace = format!("{}.{}", TEST_KEYSPACE, TEST_TABLE);
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 0, 'val0', 'static0');",
+                    table_name_under_keyspace
+                ),
+                (),
+            )
+            .await?;
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 1, 'val1', 'static1');",
+                    table_name_under_keyspace
+                ),
+                (),
+            )
+            .await?;
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 2, 'val2', 'static2');",
+                    table_name_under_keyspace
+                ),
+                (),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    async fn get_cdc_stream_id(session: &Arc<Session>) -> Result<StreamID, Box<dyn Error>> {
+        let table_name_under_keyspace = format!("{}.{}", TEST_KEYSPACE, TEST_TABLE);
+        let query_stream_id = format!(
+            "SELECT \"cdc$stream_id\" FROM {}_scylla_cdc_log LIMIT 1;",
+            table_name_under_keyspace
+        );
+
+        let mut rows = session
+            .query_iter(query_stream_id, &[])
+            .await?
+            .into_typed::<StreamID>();
+
+        Ok(rows.next().await.unwrap()?)
+    }
+
+    async fn delete_test_data_in_db(session: &Session) -> Result<(), Box<dyn Error>> {
+        session
+            .query(format!("DROP KEYSPACE IF EXISTS {}", TEST_KEYSPACE), ())
+            .await?;
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn fetch_cdc_data() -> Result<(), Box<dyn Error>> {
```

Comment:
I don't understand what the tests `fetch_cdc_data` and `fetch_with_set_upper_timestamp` are supposed to check. Could you explain?